src/main/java/org/cbio/gdcpipeline/listener/MutationStepListener.java

```diff
@@ -81,16 +81,19 @@ public ExitStatus afterStep(StepExecution stepExecution) {
                     return new ExitStatus("CONTINUE");
                 }
             }
+            //delete temp directory
+            CommonDataUtil.deleteTempDir();
         }
         return ExitStatus.COMPLETED;
     }
 
     public List<File> getMutationFileList() {
         List<File> fileList = CommonDataUtil.getFileList(gdcFileMetadatas, CommonDataUtil.GDC_TYPE.MUTATION, sourceDir);
-        List<File> mutationFileList = new ArrayList<>();
-        for (File file : fileList) {
-            File maf =new File(sourceDir,file.getName().replace(".gz", ""));
-            mutationFileList.add(maf);
+        List<File> mutationFileList = null;
+        try {
+            mutationFileList = CommonDataUtil.extractCompressedFiles(fileList);
```

Comment:
I kind of feel like `getFileList` should automatically extract the files. You shouldn't have to make two calls into the `CommonDataUtil`.

Reply:
Then either I make the extractCompressedFile call dependent on Hits parameter or I add extra logic to check inside getFileList to see if a file is compressed and extract it accordingly. What do you think ?